ramalama/common.py

```diff
@@ -205,24 +217,44 @@ def get_gpu():
 
         i += 1
 
-    if gpu_bytes:  # this is the ROCm/AMD case
-        return "HIP_VISIBLE_DEVICES", gpu_num
-
-    if os.path.exists('/etc/os-release'):
-        with open('/etc/os-release', 'r') as file:
-            content = file.read()
-            if "asahi" in content.lower():
-                return "ASAHI_VISIBLE_DEVICES", 1
-
-    return None, None
-
+    if gpu_bytes:
+        gpu_template.append({"index":gpu_num, "vram":(gpu_bytes / (1024 * 1024)), "env":"HIP_VISIBLE_DEVICES"})
+    
+    # CUDA CASE
+    nvgpu_num = 0
+    nvgpu_vram = 0
+    try:
+        command = ['nvidia-smi', '--query-gpu=index,memory.total', '--format=csv,noheader,nounits']
+        output = run_cmd(command).stdout.decode("utf-8")
+
+        # Find Nvidia GPU with largest Vram
+        for line in output.strip().split('\n'):
+            index, memory_mib = line.split(',')
+            memory_mib = int(memory_mib)
+            if memory_mib > 1024 and memory_mib > nvgpu_vram:
+                nvgpu_vram = memory_mib
+                nvgpu_num = index.strip()
+        gpu_template.append({"index":nvgpu_num, "vram":nvgpu_vram, "env":"CUDA_VISIBLE_DEVICES"})
+    except subprocess.CalledProcessError as e:
+        pass
+    except Exception as ex:
+        print(f"error occurred while detecting Nvidia gpu: {ex}")
+
+    # Select GPU
+    if gpu_template:
+        if len(gpu_template) == 1:
+            best_gpu = gpu_template[0]  # Directly use the single entry
+        else:
+            best_gpu = max(gpu_template, key=lambda x: x["vram"])  # Use max for multiple entries
+        # Set env var of picked gpu
+        os.environ[best_gpu["env"]] = best_gpu["index"]
```

Comment:
If CUDA_VISIBLE_DEVICES is set we should skip all this somehow also, can't remember how we did that for ROCm or if we did it correctly. nvidia-smi is probably an expensive technique for nvidia detection, maybe this is a start and we can do somthing cheaper later.

Make sure if nvidia-smi doesn't exists it's super silent, our packagers will complain that ramalama is broken otherwise and needs more dependancies.